src/irmin/tree.ml

```diff
@@ -198,12 +198,6 @@ module Make (P : S.PRIVATE) = struct
         match v with Hash (_, k) -> (Some k, None) | Value v -> (None, Some v)
       in
       let info = { hash; value } in
-      let v =
-        match (v, info.value, info.hash) with
-        | Value _, Some v, _ -> Value v
-        | Hash (r, _), _, Some h -> Hash (r, h)
-        | _ -> v
-      in
```

Comment:
it was actually intended to improve sharing of contents values. But I'm not sure it's worth it.

Reply:
I don't understand what you mean. I don't see any difference in behaviour with or without this statement.

Reply:
I also think it's a no-op. Looks like it might have become that way when ephemerons were removed: https://github.com/mirage/irmin/commit/f104e143af70649f7919dd47c13eca8f1a29b5f1

Reply:
Indeed, it's leftovers from an old time ;-)

Good catch!